File: 1031-add-to-array-form-of-integer/add-to-array-form-of-integer.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        # s = ""
        # for x in num:
        #     s = s + str(x)
        # result = int(s) + k  --> int(s) till 4300 digits after it overflowa
        # lst = []
        # while(result):
        #     y = result%10
        #     lst.append(y)
        #     result = result//10
        # return lst[::-1]

        i = len(num)-1
        result = []
        while i>=0 or k>0:
            if i>=0:
                k += num[i]
                i -= 1
            result.append(k%10)
            k //= 10 
        return result[::-1]

File: 1031-add-to-array-form-of-integer/add-to-array-form-of-integer.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> ls = new ArrayList<>();

        int carry = 0;
        for(int i=num.length-1; i>=0; i--){
            int sum = num[i] + k%10;
            sum += carry;
            ls.add(sum%10);
            carry = sum/10;
            k = k/10;
        }
        while(k>0){
            int sum = k%10 + carry;
            ls.add(sum%10);
            carry = sum/10;
            k = k/10;
        }
        if(carry != 0) ls.add(carry);

        Collections.reverse(ls);

        return ls;
    }
}